src/engine/GrbMatrix.h

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Johannes Herrmann (johannes.roland.herrmann@mars.uni-freiburg.de)
+
+#pragma once
+
+extern "C" {
+#include <GraphBLAS.h>
+}
+
+#include <memory>
+#include <vector>
+
+// This class wraps the functionality of the GraphBLAS object GrB_Matrix.
+// Currently only boolean matrices are supported.
+class GrbMatrix {
+ private:
+  std::unique_ptr<GrB_Matrix> matrix_;
+  static bool isInitialized_;
+
+ public:
+  GrbMatrix(size_t numRows, size_t numCols) {
+    matrix_ = std::make_unique<GrB_Matrix>();
+    auto info = GrB_Matrix_new(matrix_.get(), GrB_BOOL, numRows, numCols);
+    handleError(info);
+  }
+
+  GrbMatrix() { matrix_ = std::make_unique<GrB_Matrix>(); }
+
+  // Move constructor
+  GrbMatrix(GrbMatrix&& otherMatrix) {
+    matrix_ = std::move(otherMatrix.matrix_);
+  };
+
+  // Disable copy constructor and assignment operator
+  GrbMatrix(const GrbMatrix&) = delete;
+  GrbMatrix& operator=(const GrbMatrix&) = delete;
+
+  ~GrbMatrix() { GrB_Matrix_free(matrix_.get()); }
```

Comment:
The even cleaner way is to give a custom deleter to the `unique_ptr` (see cppreference) that does this, then your destructor is also the default (as it should be whenever possible).

Reply:
I tried to use a custom deleter, but then the function `std::make_unique` isn't usable anymore. I've checked the reference, but I'm not sure how to get the C pointer `GrB_Matrix*` properly in order to pass it to the unique_ptr constructor